src/lib/y2partitioner/dialogs/partition_move.rb

```diff
@@ -0,0 +1,125 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "cwm"
+require "y2partitioner/dialogs/popup"
+
+module Y2Partitioner
+  module Dialogs
+    # Dialog for moving a partition
+    class PartitionMove < Popup
+      # @return [Y2Storage::Partition]
+      attr_reader :partition
+
+      # @return [Symbol] :forward, :backward, :both
+      attr_reader :possible_movement
+
+      # @return [Symbol] :forward, :backward
+      attr_accessor :selected_movement
+
+      # Constructor
+      #
+      # @param partition [Y2Storage::Partition]
+      # @param possible_movement [Symbol]
+      def initialize(partition, possible_movement)
+        textdomain "storage"
+
+        @partition = partition
+        @possible_movement = possible_movement
+        @selected_movement = possible_movement
+      end
+
+      # Layout is different to the default Popup layout
+      def layout
+        VBox(
+          MarginBox(
+            2,
+            0.4,
+            ReplacePoint(Id(:contents), Empty())
+          ),
+          VSpacing(1),
+          ButtonBox(*buttons)
+        )
+      end
+
+      def contents
+        case possible_movement
+        when :both
+          VBox(DirectionSelector.new(self))
+        when :backward
+          # TRANSLATORS: %{name} is replaced by a partition name (e.g. /dev/sda1)
+          VBox(Label(format(_("Move partition %{name} backward?"), name: partition.name)))
+        when :forward
+          # TRANSLATORS: %{name} is replaced by a partition name (e.g. /dev/sda1)
+          VBox(Label(format(_("Move partition %{name} forward?"), name: partition.name)))
+        end
+      end
+
+    private
+
+      # Only shows ok and cancel buttons (help is excluded)
```

Comment:
why help is excluded?

Reply:
It is not shown in the old partitioner.

Reply:
ok, so please document it. ( and to be honest here for forward/backward help would be nice ).